packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+  // reference?: Record<string, string>;
+  reference?: Record<string, string | object>;
+};
+
+function slimDiagnosticReport(res: DiagnosticReport): SlimDiagnosticReport | undefined {
+  const mostDescriptiveType = getLongestDisplay(res.code);
+  const allTypes = getUniqueDisplays(res.code);
+
+  if (allTypes) {
+    for (const type of allTypes) {
+      if (
+        REPORT_TYPES_BLACKLIST.includes(type) ||
+        REPORT_TYPES_BLACKLIST.some(blacklistedType => type.includes(blacklistedType))
+      ) {
+        return undefined;
+      }
+    }
+  }
+
+  const uniqueData = new Set<string>();
+  let reportType = mostDescriptiveType;
+
+  res.presentedForm?.forEach(form => {
+    if (form.data) {
+      const rawData = Buffer.from(form.data, "base64").toString("utf-8");
+      const cleanedData = cleanUpNote(rawData);
+      uniqueData.add(cleanedData);
+
+      if (cleanedData.toLowerCase().includes("appointment scheduling letter")) {
+        reportType = SCHEDULING_CALL;
+      } else if (
+        cleanedData.toLowerCase().includes("unable to reach") ||
+        cleanedData.toLowerCase().includes("left a message")
+      ) {
+        reportType = UNANSWERED_CALL;
+      } else if (cleanedData.toLowerCase().includes("administrative note")) {
+        reportType = ADMIN_NOTE;
+      } else if (
+        cleanedData.toLowerCase().includes("nonva note") &&
+        cleanedData.toLowerCase().includes("refer to scanned")
+      ) {
+        reportType = SCAN_REF_NOTE;
+      } else if (cleanedData.toLowerCase().includes("discharge summary")) {
+        reportType = "discharge summary";
+      }
+    }
+  });
+
+  if (reportType && REPORT_TYPES_BLACKLIST.includes(reportType)) {
+    return undefined;
+  }
+
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  const category = res.category
+    ?.map(cat => cat.coding?.flatMap(coding => coding.display || []))
+    .join(", ");
+
+  const status = isUselessStatus(res.status) ? undefined : (res.status as string);
+
+  return {
+    ...updRes,
+    type: reportType,
+    status,
+    category,
+    presentedForm: Array.from(uniqueData),
+  };
+}
+
+type ReferenceRange = {
+  low: string | undefined;
+  high: string | undefined;
+};
+
+export type SlimObservation = Omit<
+  Observation,
+  "category" | "reading" | "value" | "interpretation" | "referenceRange" | "status" | "code"
+> & {
+  category?: string | undefined;
+  reading?: string | undefined;
+  value?: string | undefined;
+  status?: string | undefined;
+  interpretation?: string | undefined;
+  referenceRange?: ReferenceRange[] | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimObservation(res: Observation): SlimObservation {
+  const updRes = cloneDeep(res);
+  const refRange = res.referenceRange?.map(range => ({
+    low: getQuantityString(range.low),
+    high: getQuantityString(range.high),
+  }));
+  const status = isUselessStatus(res.status) ? undefined : (res.status as string);
```

Comment:
Why the casting to `string`?

Reply:
just cuz the FHIR DiagReport status doesnt have an easy status type to copy.. it's like this:
```  status?: 'registered' | 'partial' | 'preliminary' | 'final' | 'amended' | 'corrected' | 'appended' | 'cancelled' | 'entered-in-error' | 'unknown';```

So i decided we just cast it as string | undefined to make it simpler

Reply:
Not to drag since this is NABD, but wouldn't it be fine to have the `status` const as an union type instead of `string`?